spec/reek/smells/instance_variable_assumption_spec.rb

```diff
@@ -0,0 +1,207 @@
+require_relative '../../spec_helper'
+
+require_lib 'reek/smells/instance_variable_assumption'
+
+RSpec.describe Reek::Smells::InstanceVariableAssumption do
+  describe 'warning' do
+    context 'smell line' do
+      it 'should report the lines' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, lines: [1])
+      end
+    end
+
+    context 'smell parameters' do
+      it 'should report the lines' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, assumptions: [:@a])
+      end
+    end
+
+    context 'smell context' do
+      it 'should report the context' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, context: 'Dummy')
+      end
+    end
+
+    context 'smell message' do
+      it 'should report the ivars in the message' do
+        message = 'assumes too much for instance variable @a, @b'
+
+        src = <<-EOS
+          class Dummy
+            def test
+              [@a, @b]
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, message: message)
+      end
+
+      it 'should report each ivar once' do
+        message = 'assumes too much for instance variable @a, @b, @c'
+
+        src = <<-EOS
+          class Dummy
+            def test
+              [@a, @a, @b, @c]
+            end
+
+            def retest
+              @c
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, message: message)
+      end
+    end
+  end
+
+  it 'should not report an empty class' do
+    src = <<-EOS
+      class Dummy
+      end
+    EOS
+
+    expect(src).not_to reek_of(:InstanceVariableAssumption)
+  end
+
+  it 'should not report when lazy initializing' do
+    src = <<-EOS
+      class Dummy
+        def test
+          @a ||= 1
+        end
+      end
+    EOS
+
+    expect(src).not_to reek_of(:InstanceVariableAssumption)
+  end
+
+  it 'should report when making instance variable assumption' do
+    src = <<-EOS
+      class Dummy
+        def test
+          @a
+        end
+      end
+    EOS
+
+    expect(src).to reek_of(:InstanceVariableAssumption)
+  end
+
+  it 'should report any initialization outside initialize' do
```

Comment:
Maybe 'reports variable even if others are initialized' describes better what is being tested here?